### PR TITLE
Improve performance of controls/selected page

### PR DESCRIPTION
**Pull request by woodt**
- Rewrite Catalogs class to leverage cached Catalog instances
- Rewrite component count query
- Fix unrelated bug causing crash when displaying component details
  for components w/o any statements

### CODE REVIEW

File: controls/views.py [hunk 1]
```diff
@@ -1340,11 +1356,13 @@ def component_library_component(request, element_id):
...
             "enable_experimental_opencontrol": SystemSettings.enable_experimental_opencontrol,
             "form_source": "component_library",
-            "inheritance": inheritance,
+            "inheritances": inheritances,
         }
         return render(request, "components/element_detail_tabs.html", context)
 
+    
     if len(impl_smts) == 0:
+        # TODO: not reached given previous if statement
```
Comment (Tom-Camp) [inline controls/views.py]: We should just remove this condition.

**Review by Tom-Camp — CHANGES_REQUESTED** [review]
Let's remove the `if len(impl_smts) == 0:` condition.

File: controls/views.py (hunk 1 above)
Comment (woodt) [reply controls/views.py]: Removed!

**Review by Tom-Camp — APPROVED** [review]

**Comment by Tom-Camp** [comment]
Looks good, Tom. Thank you.

**Pull request merged**